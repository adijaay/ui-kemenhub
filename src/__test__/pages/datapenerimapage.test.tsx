import DataPenerimaPage from "@/pages/data-penerima";
import "@testing-library/jest-dom";
import { render, waitFor } from "@testing-library/react";
import { useRouter } from "next/router";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

jest.mock("@/components/ui/Page", () => {
  return jest.fn().mockImplementation(({ children }) => {
    return <div>{children}</div>;
  });
});

jest.mock("@/utils/firebase", () => ({
  analytics: jest.fn(),
  logEvent: jest.fn(),
}));

describe("DataPenerimaPage", () => {
  (useRouter as jest.Mock).mockReturnValue({
    query: {},
  });

  it("render data penerima page correctly", async () => {
    const page = render(<DataPenerimaPage />);
    await waitFor(() => {
      expect(page).toMatchSnapshot();
    });
  });
});
