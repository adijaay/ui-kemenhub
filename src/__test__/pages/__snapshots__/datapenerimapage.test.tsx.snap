// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataPenerimaPage render data penerima page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 !p-4 !border-[#FEDF88] !bg-[#FFFCF5]"
        >
          <div
            class="flex gap-3"
          >
            <svg
              class="tabler-icon tabler-icon-alert-circle shrink-0"
              fill="none"
              height="24"
              stroke="#F79008"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 12a9 9 0 1 0 18 0a9 9 0 0 0 -18 0"
              />
              <path
                d="M12 8v4"
              />
              <path
                d="M12 16h.01"
              />
            </svg>
            <div
              class="flex"
            >
              <div
                class="flex w-full flex-col gap-1"
              >
                <p
                  class="font-semibold text-[#1f1f1f] text-sm"
                >
                  Segera Aktivasi Rekening Anda
                </p>
                <p
                  class="text-sm font-normal text-[#1f1f1f] text-xs leading-[18px]"
                >
                  Lakukan aktivasi rekening untuk mendapatkan bantuan dana.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 !gap-0 !p-0"
        >
          <div
            class="border-b border-[#EDEDED] px-4 py-3"
          >
            <p
              class="text-sm font-semibold text-[#1f1f1f]"
            >
              Data Diri Penerima
            </p>
          </div>
          <div
            class="flex flex-col gap-2 px-4 py-2"
          >
            <p
              class="text-xs font-normal leading-5 text-[#667085]"
            >
              Nama Lengkap
            </p>
            <p
              class="text-sm font-semibold text-[#1F1F1F]"
            />
          </div>
          <div
            class="flex flex-col gap-2 px-4 py-2"
          >
            <p
              class="text-xs font-normal leading-5 text-[#667085]"
            >
              NIK
            </p>
            <p
              class="text-sm font-semibold text-[#1F1F1F]"
            />
          </div>
          <div
            class="flex flex-col gap-2 px-4 py-2"
          >
            <p
              class="text-xs font-normal leading-5 text-[#667085]"
            >
              NISN
            </p>
            <p
              class="text-sm font-semibold text-[#1F1F1F]"
            />
          </div>
        </div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 !gap-0 !p-0"
        >
          <div
            class="flex items-center justify-between rounded-t-2xl border-b border-[#EAECF0] bg-white p-4"
          >
            <div
              class="flex flex-col gap-4"
            >
              <p
                class="font-semibold text-[#1f1f1f] text-sm"
              >
                Jenjang 
                SMP
              </p>
              <div
                class="flex max-w-[162.5px] flex-col gap-1"
              >
                <p
                  class="text-xs text-[#667085]"
                >
                  Sekolah
                </p>
                <p
                  class="text-sm font-semibold text-[#1f1f1f]"
                >
                  SMP Penabur Esa Unggul
                </p>
              </div>
            </div>
            <div
              class="rounded-full px-2 py-1 bg-[#FFFAEA] text-[#F79008]"
            >
              <p
                class="text-sm font-medium leading-5"
              >
                Perlu Aktivasi
              </p>
            </div>
          </div>
          <div
            class="px-4 py-3"
          >
            <div
              class="flex items-center justify-between"
            >
              <div
                class="flex w-[162.5px] flex-col gap-1"
              >
                <p
                  class="text-xs leading-5 text-[#667085]"
                >
                  Rekening Penerima
                </p>
                <p
                  class="text-sm font-normal text-[#333333]"
                >
                  223 2881 1921 121
                </p>
              </div>
              <button
                class="flex h-10 items-center gap-1"
                type="button"
              >
                <span
                  class="text-xs font-semibold leading-5 text-[#1F1F1F]"
                >
                  Lihat Riwayat
                </span>
                <svg
                  class="tabler-icon tabler-icon-chevron-down rotate-0 duration-300"
                  fill="none"
                  height="16"
                  stroke="#667085"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 9l6 6l6 -6"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 !gap-0 !p-0"
        >
          <div
            class="flex items-center justify-between rounded-t-2xl border-b border-[#EAECF0] bg-white p-4"
          >
            <div
              class="flex flex-col gap-4"
            >
              <p
                class="font-semibold text-[#1f1f1f] text-sm"
              >
                Jenjang 
                SD
              </p>
              <div
                class="flex max-w-[162.5px] flex-col gap-1"
              >
                <p
                  class="text-xs text-[#667085]"
                >
                  Sekolah
                </p>
                <p
                  class="text-sm font-semibold text-[#1f1f1f]"
                >
                  SD Harapan Bangsa
                </p>
              </div>
            </div>
            <div
              class="rounded-full px-2 py-1 bg-[#ECFEF3] text-[#19B26B]"
            >
              <p
                class="text-sm font-medium leading-5"
              >
                Selesai
              </p>
            </div>
          </div>
          <div
            class="px-4 py-3"
          >
            <div
              class="flex items-center justify-between"
            >
              <div
                class="flex w-[162.5px] flex-col gap-1"
              >
                <p
                  class="text-xs leading-5 text-[#667085]"
                >
                  Rekening Penerima
                </p>
                <p
                  class="text-sm font-normal text-[#333333]"
                >
                  331 0012 2291 221
                </p>
              </div>
              <button
                class="flex h-10 items-center gap-1"
                type="button"
              >
                <span
                  class="text-xs font-semibold leading-5 text-[#1F1F1F]"
                >
                  Lihat Riwayat
                </span>
                <svg
                  class="tabler-icon tabler-icon-chevron-down rotate-0 duration-300"
                  fill="none"
                  height="16"
                  stroke="#667085"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 9l6 6l6 -6"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 !p-4 !border-[#FEDF88] !bg-[#FFFCF5]"
      >
        <div
          class="flex gap-3"
        >
          <svg
            class="tabler-icon tabler-icon-alert-circle shrink-0"
            fill="none"
            height="24"
            stroke="#F79008"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12a9 9 0 1 0 18 0a9 9 0 0 0 -18 0"
            />
            <path
              d="M12 8v4"
            />
            <path
              d="M12 16h.01"
            />
          </svg>
          <div
            class="flex"
          >
            <div
              class="flex w-full flex-col gap-1"
            >
              <p
                class="font-semibold text-[#1f1f1f] text-sm"
              >
                Segera Aktivasi Rekening Anda
              </p>
              <p
                class="text-sm font-normal text-[#1f1f1f] text-xs leading-[18px]"
              >
                Lakukan aktivasi rekening untuk mendapatkan bantuan dana.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 !gap-0 !p-0"
      >
        <div
          class="border-b border-[#EDEDED] px-4 py-3"
        >
          <p
            class="text-sm font-semibold text-[#1f1f1f]"
          >
            Data Diri Penerima
          </p>
        </div>
        <div
          class="flex flex-col gap-2 px-4 py-2"
        >
          <p
            class="text-xs font-normal leading-5 text-[#667085]"
          >
            Nama Lengkap
          </p>
          <p
            class="text-sm font-semibold text-[#1F1F1F]"
          />
        </div>
        <div
          class="flex flex-col gap-2 px-4 py-2"
        >
          <p
            class="text-xs font-normal leading-5 text-[#667085]"
          >
            NIK
          </p>
          <p
            class="text-sm font-semibold text-[#1F1F1F]"
          />
        </div>
        <div
          class="flex flex-col gap-2 px-4 py-2"
        >
          <p
            class="text-xs font-normal leading-5 text-[#667085]"
          >
            NISN
          </p>
          <p
            class="text-sm font-semibold text-[#1F1F1F]"
          />
        </div>
      </div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 !gap-0 !p-0"
      >
        <div
          class="flex items-center justify-between rounded-t-2xl border-b border-[#EAECF0] bg-white p-4"
        >
          <div
            class="flex flex-col gap-4"
          >
            <p
              class="font-semibold text-[#1f1f1f] text-sm"
            >
              Jenjang 
              SMP
            </p>
            <div
              class="flex max-w-[162.5px] flex-col gap-1"
            >
              <p
                class="text-xs text-[#667085]"
              >
                Sekolah
              </p>
              <p
                class="text-sm font-semibold text-[#1f1f1f]"
              >
                SMP Penabur Esa Unggul
              </p>
            </div>
          </div>
          <div
            class="rounded-full px-2 py-1 bg-[#FFFAEA] text-[#F79008]"
          >
            <p
              class="text-sm font-medium leading-5"
            >
              Perlu Aktivasi
            </p>
          </div>
        </div>
        <div
          class="px-4 py-3"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex w-[162.5px] flex-col gap-1"
            >
              <p
                class="text-xs leading-5 text-[#667085]"
              >
                Rekening Penerima
              </p>
              <p
                class="text-sm font-normal text-[#333333]"
              >
                223 2881 1921 121
              </p>
            </div>
            <button
              class="flex h-10 items-center gap-1"
              type="button"
            >
              <span
                class="text-xs font-semibold leading-5 text-[#1F1F1F]"
              >
                Lihat Riwayat
              </span>
              <svg
                class="tabler-icon tabler-icon-chevron-down rotate-0 duration-300"
                fill="none"
                height="16"
                stroke="#667085"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 9l6 6l6 -6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 !gap-0 !p-0"
      >
        <div
          class="flex items-center justify-between rounded-t-2xl border-b border-[#EAECF0] bg-white p-4"
        >
          <div
            class="flex flex-col gap-4"
          >
            <p
              class="font-semibold text-[#1f1f1f] text-sm"
            >
              Jenjang 
              SD
            </p>
            <div
              class="flex max-w-[162.5px] flex-col gap-1"
            >
              <p
                class="text-xs text-[#667085]"
              >
                Sekolah
              </p>
              <p
                class="text-sm font-semibold text-[#1f1f1f]"
              >
                SD Harapan Bangsa
              </p>
            </div>
          </div>
          <div
            class="rounded-full px-2 py-1 bg-[#ECFEF3] text-[#19B26B]"
          >
            <p
              class="text-sm font-medium leading-5"
            >
              Selesai
            </p>
          </div>
        </div>
        <div
          class="px-4 py-3"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex w-[162.5px] flex-col gap-1"
            >
              <p
                class="text-xs leading-5 text-[#667085]"
              >
                Rekening Penerima
              </p>
              <p
                class="text-sm font-normal text-[#333333]"
              >
                331 0012 2291 221
              </p>
            </div>
            <button
              class="flex h-10 items-center gap-1"
              type="button"
            >
              <span
                class="text-xs font-semibold leading-5 text-[#1F1F1F]"
              >
                Lihat Riwayat
              </span>
              <svg
                class="tabler-icon tabler-icon-chevron-down rotate-0 duration-300"
                fill="none"
                height="16"
                stroke="#667085"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 9l6 6l6 -6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
